src/hooks/useFetch.js

```diff
@@ -0,0 +1,60 @@
+import { useState, useEffect } from "react";
+
+export const useFetch = (url, method = "GET") => {
+  const [data, setData] = useState(null);
+  const [isPending, setIsPending] = useState(false);
+  const [error, setError] = useState(null);
+  const [options, setOptions] = useState(null);
+
+  const postData = (postData) => {
+    setOptions({
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(postData),
+    });
+  };
+
+  useEffect(() => {
+    const controller = new AbortController();
+
+    const fetchData = async (fetchOptions) => {
```

Comment:
I like you've created an ```fetchOptions``` parameter so that the request function could be in one!